### DISCUSSION TITLE
Question about MIDI Type 2

### DISCUSSION THREAD

**Discussion by Jenix-Park** [melanchall/drywetmidi#248] [Q&A] [answered]
Hello,

I would like to ask for your advice on whether or not to support MIDI Type 2 files in a MIDI player app that I am currently developing. While I understand that MIDI Type 2 may be useful for some professionals in specific cases, I am curious to know if you have ever come across MIDI Type 2 files in general usage, such as those floating around the internet.

I appreciate any guidance you can provide on this matter.

And, as always, thank you for this great library.

**Comment by melanchall** [comment, marked as answer]
Hi,

My opinion is you can safely think that there are no such files in the wild nature. I have some sample files of Type 2 format [here](https://github.com/melanchall/drywetmidi/tree/develop/Resources/MIDI%20files/Valid/MultiSequence), but I found them many years ago and just used them to check that DryWetMIDI can read them.

For almost six years of the library life I've never received a message or issue related to such files. Maybe [midi.org forum](https://midi.org/forum) is a better place where you can ask this question. But I think you can forget about Type 2 files.

Thanks,
Max

**Reply by Jenix-Park** [reply]
Thank you for the clarification!
